api/apiFunctions.test.tsx

```diff
@@ -24,294 +22,370 @@ jest.mock('./apiFunctions', () => {
+        'Error getting all leagues',
+      );
+    });
   });
-});
 
-describe('add user to league', () => {
-  const response = {
-    userDocumentId: 'user123',
-    selectedLeague: 'league123',
-    selectedLeagues: ['league123'],
-    participants: ['user123', 'user456'],
-    survivors: ['user123'],
-  };
-  it('should add user to league', async () => {
-    apiFunctions.addUserToLeague.mockResolvedValue(response);
-    const result = await apiFunctions.addUserToLeague({ response });
-    expect(result).toEqual(response);
-  });
-  it('should return error upon unsuccessful call', async () => {
-    apiFunctions.addUserToLeague.mockRejectedValue(
-      new Error('Error adding user to league'),
-    );
-    await expect(apiFunctions.addUserToLeague('123', '456')).rejects.toThrow(
-      'Error adding user to league',
-    );
+  describe('get users id', () => {
+    it('should return user document ID', async () => {
+      const userId = '123';
+      const response = '456';
+      apiFunctions.getUserDocumentId.mockResolvedValue(response);
+      const result = await apiFunctions.getUserDocumentId(userId);
+      expect(result).toEqual(response);
+    });
+    it('should return error upon unsuccessful call', async () => {
+      apiFunctions.getUserDocumentId.mockRejectedValue(
+        new Error('Error getting user document ID'),
+      );
+      await expect(apiFunctions.getUserDocumentId('123')).rejects.toThrow(
+        'Error getting user document ID',
+      );
+    });
   });
-  it('should return error upon unsuccessful call', async () => {
-    apiFunctions.addUserToLeague.mockRejectedValue(
-      new Error('Error adding user to league'),
-    );
-    await expect(apiFunctions.addUserToLeague({ response })).rejects.toThrow(
-      'Error adding user to league',
-    );
+
+  describe('add user to league', () => {
+    const response = {
+      userDocumentId: 'user123',
+      selectedLeague: 'league123',
+      selectedLeagues: ['league123'],
+      participants: ['user123', 'user456'],
+      survivors: ['user123'],
+    };
+    it('should add user to league', async () => {
+      apiFunctions.addUserToLeague.mockResolvedValue(response);
+      const result = await apiFunctions.addUserToLeague({ response });
+      expect(result).toEqual(response);
+    });
+    it('should return error upon unsuccessful call', async () => {
+      apiFunctions.addUserToLeague.mockRejectedValue(
+        new Error('Error adding user to league'),
+      );
+      await expect(apiFunctions.addUserToLeague('123', '456')).rejects.toThrow(
+        'Error adding user to league',
+      );
+    });
+    it('should return error upon unsuccessful call', async () => {
+      apiFunctions.addUserToLeague.mockRejectedValue(
+        new Error('Error adding user to league'),
+      );
+      await expect(apiFunctions.addUserToLeague({ response })).rejects.toThrow(
+        'Error adding user to league',
+      );
+    });
```

Comment:
It seems the only difference between these two tests are you are adding response instead of hard-coded values, do we need two of these?

Reply:
✅ good catch! fixed.